### PR TITLE
🏗 Convert all OWNERS.yaml files to JSON5 format

### PR DESCRIPTION
**Pull request by rcebulko**
Created with [this conversion script](https://gist.github.com/rcebulko/1bdc4086f8bf3e97a3119d0d9b7cf95e)

To generate this PR, each OWNERS.yaml file was parsed and then re-written in JSON5 format. The original YAML files were deleted. The accuracy of this transition was verified by parsing the new vs old OWNERS file sets and verifying that the output ownership tree was identical.

This should be submitted as soon as possible to reduce risk of merge conflicts. It's better to do it like this (rather than leave both YAML and JSON versions in place) so there is less risk of contributors editing the deprecated YAML files. Note that the parser will still support YAML (ie. it won't break anything if someone checks in an old YAML file), and I'm still being notified of all OWNERS/OWNERS.yaml changes, so I can catch anyone who tries :policeman: 

The new owners files point to the updated example OWNERS file.

Related to https://github.com/ampproject/amp-github-apps/issues/442

### CODE REVIEW

File: OWNERS [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      pattern: '*.md',
+      owners: [
+        { name: 'ampproject/wg-outreach' }
+      ]
+    },
+
+    {
+      pattern: '{.*,babel.config.js,gulpfile.js}',
+      owners: [
+        { name: 'ampproject/wg-infra' }
```
Comment (rsimha) [inline OWNERS]: I noticed that the new json5 files have initially been auto-formatted w.r.t. indentation, new lines, etc. As people manually edit `OWNERS` files from here on, they could break the formatting convention and / or introduce parsing errors. Would it be possible to have `prettier` auto-format these files like other `.js` files?

File: OWNERS [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      // The below rule will be here only temporarily during the transition
+      // period while many owners files are being updated as contributors and
+      // teams notice which files are out of date. The "notify" option means the
+      // owners bot will tag me on any PR that touches an owners file. This will
+      // allow me to verify syntax and semantics of all updates until the syntax
+      // check API for Travis to use is complete.
+
+      // Though this gives me the ability to approve changes to owners files,
+      // I do not currently have merge access; the owners check is currently
+      // non-blocking; and I run and maintain the owners bot application.
```
Comment (rsimha) [inline OWNERS]: nit: For more clarity, I suggest changing this paragraph from first person to third person (referencing `@rcebulko`).

File: OWNERS (hunk 1 above)
Comment (rcebulko) [reply OWNERS]: Filed tracking issue #24936 to address in follow-up PR (since this PR is already massive)

File: OWNERS (hunk 2 above)
Comment (rcebulko) [reply OWNERS]: Done

**Comment by rcebulko** [comment]
For posterity: This PR will be submitted with a failing owners check because it modifies owners files without extensions, but I have ownership of all owners files and it's a no-op change that does not actually change any rules.

File: OWNERS (hunk 1 above)
Comment (rsimha) [reply OWNERS]: All `OWNERS` files were auto formatted in #25085.

**Pull request merged**